addons/mass_mailing_event/models/event_registration.py

```diff
@@ -9,4 +11,8 @@ class EventRegistration(models.Model):
     _mailing_enabled = True
 
     def _mailing_get_default_domain(self, mailing):
-        return [('state', 'not in', ['cancel', 'draft'])]
+        default_domain = [('state', 'not in', ['cancel', 'draft'])]
+        default_mailing_model_id = self.env.context.get('mailing_model_id')
+        if mailing.mailing_model_id.id == default_mailing_model_id and self.env.context.get('mailing_domain'):
```

Comment:
Use "default_mailing_domain" as first check (in case it works again), then "force_mailing_domain" as fallback key (to avoid context keys named like fields, it eases grepping them).